crates/pixi_command_dispatcher/src/build/dependencies.rs

```diff
@@ -200,24 +204,30 @@ impl Dependencies {
                 .collect()
         }
 
-        for record in records {
+        let mut filtered_records = records
+            .iter_mut()
             // Only record run exports for packages that are direct dependencies.
-            if !self
-                .dependencies
-                .contains_key(&record.package_record().name)
-            {
-                continue;
-            }
-
+            .filter(|r| !self.dependencies.contains_key(&r.package_record().name))
             // Filter based on whether we want to ignore run exports for a particular
             // package.
-            if ignore.from_package.contains(&record.package_record().name) {
-                continue;
-            }
+            .filter(|r| ignore.from_package.contains(&r.package_record().name))
```

Comment:
This should also be inverted! We dont want any package that should be ignored.

```suggestion
            .filter(|r| !ignore.from_package.contains(&r.package_record().name))
```